Source/Renderer/modernizeShader.js

```diff
@@ -99,21 +99,23 @@ function modernizeShader(source, isFragmentShader) {
     }
   }
 
+  var webgl2UniqueID = "WEBGL_2";
+  var webgl2DefineMacro = "#define " + webgl2UniqueID;
   var versionThree = "#version 300 es";
   var foundVersion = false;
   for (i = 0; i < splitSource.length; i++) {
     if (/#version/.test(splitSource[i])) {
-      splitSource[i] = versionThree;
+      splitSource[i] = versionThree + "\n" + webgl2DefineMacro;
```

Comment:
Since #version is required to be on top, should I break the loop early?

Reply:
Yes, that should be fine